File: src/client/src/models/Holding.ts
import moment from "moment";

export interface IHolding {
    id: number;
    accountId: number;
    account_id?: number;
    securitySymbol: string;
    security_symbol?: string;
    costBasis: number;
    cost_basis?: number;
    quantity: number;
    isoCurrencyCode: string;
    iso_currency_code?: string;
    latestPrice: number;
    latest_price?: number;
    timestamp: string | Date;
}

class Holding {
    public id: number;
    public accountId: number;
    public securitySymbol: string;
    public costBasis: number;
    public quantity: number;
    public isoCurrencyCode: string;
    public latestPrice: number;
    private _timestamp: string | Date;

    public get timestamp(): Date {
        return moment(this._timestamp).toDate();
    }

    constructor(serverObj: IHolding = {} as IHolding) {
        this.id = serverObj.id;
        this.accountId = serverObj.account_id;
        this.securitySymbol = serverObj.security_symbol;
        this.costBasis = serverObj.cost_basis;
        this.quantity = serverObj.quantity;
        this.isoCurrencyCode = serverObj.iso_currency_code;
        this.latestPrice = serverObj.latest_price;
        this._timestamp = serverObj.timestamp;
    }
}

export default Holding;

File: src/core/models/holding.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Float
from sqlalchemy.orm import relationship

from core.models.base import Base


class Holding(Base):
    __tablename__ = 'holdings'

    id = Column(Integer, primary_key=True)
    account_id = Column(Integer, ForeignKey("accounts.id"), nullable=False)
    symbol = Column(String(64))
    cost_basis = Column(Float)
    quantity = Column(Float)
    iso_currency_code = Column(String(8))
    timestamp = Column(DateTime)

    account = relationship("Account", back_populates="holdings")
    balances = relationship("HoldingBalance", back_populates="holding")
